clean-data/clean-data.R

```diff
@@ -0,0 +1,700 @@
+        mutate_each(funs(grepl(findStr, ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    subDirty <- dirtyDat %>% filter(dirtyIdx) %>%
+        mutate_(.dots = setNames(list(varval), colName))
+
+    cleanDat <- dirtyDat %>% filter(!dirtyIdx) %>% bind_rows(subDirty)
+
+    cleanDat
+}
+
+# Clean Expected Earnings
+clean_expected_earnings <- function(cleanPart1) {
+    # Remove dollar signs from expected earnings
+    dollarIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("\\$", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    dollarData <- cleanPart1 %>% filter(dollarIdx) %>%
+        mutate(ExpectedEarning = sub("\\$", "", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!dollarIdx) %>% bind_rows(dollarData)
+
+    # Remove commas from expected earnings
+    commaIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl(",", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    commaData <- cleanPart1 %>% filter(commaIdx) %>%
+        mutate(ExpectedEarning = sub(",", "", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!commaIdx) %>% bind_rows(commaData)
+
+    # Remove "k" from expected earnings
+    kIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("k", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    kData <- cleanPart1 %>% filter(kIdx) %>%
+        mutate(ExpectedEarning = sub("k", "000", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!kIdx) %>% bind_rows(kData)
+
+    # Change range of expected earnings into average
+    rangeIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("-", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    rangeData <- cleanPart1 %>% filter(rangeIdx) %>%
+        mutate(ExpectedEarning = average_range_earning(ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!rangeIdx) %>% bind_rows(rangeData)
+
+    # Remove period from salaries like 50.000 which should be just 50000
+    thousandsIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("^\\d{2}\\.", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    thousandsData <- cleanPart1 %>% filter(thousandsIdx) %>%
+        mutate(ExpectedEarning = sub("\\.", "", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!thousandsIdx) %>%
+        bind_rows(thousandsData)
+
+    # Remove any non-numeric characters
+    numericIdx <- cleanPart1 %>% select(ExpectedEarning) %>%
+        mutate_each(funs(grepl("[A-Za-z]", ., ignore.case = TRUE))) %>%
+        unlist(use.names = FALSE)
+    numericData <- cleanPart1 %>% filter(numericIdx) %>%
+        mutate(ExpectedEarning = gsub("[A-Za-z']*", "", ExpectedEarning))
+    cleanPart1 <- cleanPart1 %>% filter(!numericIdx) %>%
+        bind_rows(numericData)
+
+    # Make all expected earnings less than 100 multiplied by 1000
```

Comment:
@SamAI-Software the 100 was arbitrary. I felt that making 120 to 120,000 was too much of a leap so I stuck with 100. Plus, at what point do we threshold and say above this number, we're going to stop multiplying?

What is making me scratch my head is I'm unsure as to why individuals would put values as low as 10 or so. Do you have any ideas? Thanks for the note on some countries not using annual salaries. I didn't know that and that'll help inform some decisions from here on.

I like the idea of minimum and maximum salary cutoffs. Same question to you, why have an upper bound of 6000 to multiply by 12? 😄